src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -138,6 +141,20 @@ static BoundNode unwrapListElement(BoundCollectionExpression node, BoundNode ele
             }
         }
 
+        private bool IsSingleReadOnlySpanSpread(BoundCollectionExpression node, TypeWithAnnotations elementType, [NotNullWhen(true)] out BoundExpression? spreadExpression)
+        {
+            spreadExpression = null;
+
+            if (node is { Elements: [BoundCollectionExpressionSpreadElement { Expression: { Type: NamedTypeSymbol { TypeArgumentsWithAnnotationsNoUseSiteDiagnostics: [var spreadElementType] } spreadType } expr }] } &&
+                spreadType.OriginalDefinition == (object)_compilation.GetWellKnownType(WellKnownType.System_ReadOnlySpan_T) &&
+                spreadElementType.Equals(elementType, TypeCompareKind.CLRSignatureCompareOptions))
```

Comment:
>CLRSignatureCompareOptions [](http://example.com/codeflow?start=70&length=26)

Are we testing this `TypeCompareKind`? Consider testing cases such as:
```csharp
static MyCollection<object> F1(ReadOnlySpan<dynamic> s) => [.. s];              // no copy
static MyCollection<(int A, int B)> F2(ReadOnlySpan<(int, int)> s) => [.. s];   // no copy
static MyCollection<(object, object)> F3(ReadOnlySpan<(int, int)> s) => [.. s]; // copy
static MyCollection<T?> F4<T>(ReadOnlySpan<T> s) => [.. s];                     // no copy
```

Reply:
It might be reasonable to try and use a similar check that `PrepareCopyToOptimization` is already doing.

https://github.com/dotnet/roslyn/blob/bb722db74c453c5304167be34762da69a4842492/src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs#L809-L812

See also tests added in #71195